BE/src/main/java/kr/codesquad/airbnb11/controller/ReservationInterceptor.java

```diff
@@ -0,0 +1,52 @@
+package kr.codesquad.airbnb11.controller;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import kr.codesquad.airbnb11.common.error.exception.LoginRequiredException;
+import kr.codesquad.airbnb11.domain.user.UserDTO;
+import kr.codesquad.airbnb11.service.JwtService;
+import kr.codesquad.airbnb11.service.LoginService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+public class ReservationInterceptor extends HandlerInterceptorAdapter {
+
+  private static Logger log = LoggerFactory.getLogger(ReservationInterceptor.class);
+
+  @Autowired
+  private JwtService jwtService;
+
+  @Autowired
+  private LoginService loginService;
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
+      Object handler) {
+
+    String token = "";
+    Cookie[] cookies = request.getCookies();
+
+    if (cookies == null || cookies.length < 1) {
+      throw new LoginRequiredException("로그인이 필요합니다");
+    }
+
+    for (Cookie cookie : cookies) {
+      if (cookie.getName().equals("jwt")) {
+        token = cookie.getValue();
+        log.debug("token : {}", token);
+        break;
+      }
+    }
```

Comment:
함수형을 적용해보는건 어떨까 싶어서 써봤는데 성능이 구릴 것 같네요.
```suggestion
    String token = Arrays.stream(cookies)
        .filter(c -> c.getName().equals("jwt"))
        .map(Cookie::getValue)
        .reduce(String::concat)
        .orElse("");
```

Reply:
함수형으로 바꾸는게 좀 더 보기 편한것 같아요 바꾸겠습니당!!

Reply:
아 성능이 안좋아요 ㅋㅋㅋㅋㅋㅋ 그냥 써본거에요!